### PR TITLE
Curvilinear Perspective: Add fixed-point library and asset conversion helpers

### LINKED ISSUE #452: Add fixed point helpers for BigInt numbers
Helpers specifically focused on UI and conversion:
- Support conversions from assets to main currencies.
- Support floating-point strings -> BigInt-with-fixed-point-information objects (in the same structure as internal asset numbers).
- As relevant, support for math operations between the BigInt-with-fixed-point-information objects.

### PR DESCRIPTION
**Pull request by Shadowfiend**
This implements a very lightweight fixed point math library, and uses it to
implement more complete asset conversions. The asset conversions are not yet
hooked into the places where we need them; that will happen in follow-on work.

--------

@itsrachelfish fixed-point.ts, and particularly the `FixedPointNumber` type,
`parseToFixedPointNumber`, and `multiplyFixedPointNumbers` should be of specific
interest to you, though would appreciate review of the rest of the code if you could.

@mhluongo would love sanity checks on the conversion stuff as well.

Closes #452 .

### CODE REVIEW

**Comment by Shadowfiend** [comment]
Forgot to update a price test to reflect the fixed decimals info on conversions. Rebase coming tonight.

File: background/lib/fixed-point.ts [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+  )
+}
+
+/**
+ * Converts the given `floatingPoint` number to a FixedPointNumber with
+ * `fixedPointDecimals` of precision.
+ */
+export function toFixedPointNumber(
+  floatingPoint: number,
+  fixedPointDecimals: number
```
Comment (mhluongo) [inline background/lib/fixed-point.ts:124]: Shouldn't this just be named `decimals`? V confused about the naming choices here, feels like I'm missing something

File: background/lib/fixed-point.ts [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+ *   amount: 100893000107n,
+ *   decimals: 5
+ * }
+ * ```
+ *
+ * Convenience functions exist in this file to convert regular JavaScript
+ * floating point Number to and from FixedPointNumber, as well as to multiply
+ * FixedPointNumber and floats.
+ */
+export type FixedPointNumber = {
```
[the remark this answers is not in the document]
Comment (mhluongo) [reply background/lib/fixed-point.ts:18]: Understood if you want to keep it bare for performance reasons, but if so maybe we shouldn't breaking it out so much in the util functions

File: background/lib/fixed-point.ts [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    "$1.$2"
+  )
+  const [whole, decimals] = noThousands.split(".")
+  return {
+    amount: BigInt(whole + decimals),
+    decimals: decimals.length,
+  }
+}
+
+export function fixedPointNumberToString({
```
Comment (mhluongo) [inline background/lib/fixed-point.ts:153]: Would be nice to have "humanize" and "truncate" formatting functions as well. Doesn't need to happen in this PR of course

**Review by mhluongo — APPROVED** [review]
I don't agree with all of the API choices here, but I'm down to see if this is what we need. Leaving it unmerged in case this resonates or someone else wants to share their 2 sats

**Comment by mhluongo** [comment]
Of course, I forgot about the holiday... not going to wait for another reviewer, but would love to hear your thoughts later @Shadowfiend.

Let's do this thing :ship: :shipit:

File: background/lib/fixed-point.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Shadowfiend) [reply background/lib/fixed-point.ts:18]: I batted back and forth here and went with the approach most consistent with what we already have, partly to minimize custom serialization logic and partly to have a consistent way of handling data and operations on it (simple functions) vs mediation between data provision/storage (classes and instances).

That said, I agree there's a lot of room for better naming throughout here, and the slightly scratch outcome is the result of time constraints and needing to move on in the immediate term.

The original version of this had functions that always took amount and decimals separately, but I found myself writing admonitions in the docs to make sure to track those together, at which point I introduced a simple way to do exactly that with FixedPointNumber and adapted the functions to work with that when possible. My expectation is that swaps will use that version, and I wanted to make sure it was easy to do that math safely.

**Pull request merged**